### PR TITLE
Fix min_weight_matching to convert edge weights without reciprocal

### LINKED ISSUE #5388: min_weight_matching gives incorrect results
Consider the following graph:
```python
G = nx.Graph()
G.add_edge(1, 4, weight=2)
G.add_edge(2, 3, weight=2)
G.add_edge(1, 2, weight=1)
G.add_edge(3, 4, weight=4)
nx.min_weight_matching(G)
```

### Current Behavior

The program returns `{(2, 1), (4, 3)}`
### Expected Behavior

There are two maximal matchings in the graph, (1,2)(3,4) with weight 1+4=5, and (1,4)(2,3) with weight 2+2=4. The minimum weight maximal matching should be (1,4)(2,3) which has weight 4.
### Steps to Reproduce

Run the code provided
### Environment

Python version: 3.9
NetworkX version: 2.7.1

### Additional context

From the documentation and the code, `min_weight_matching` is simply taking the reciprocal of the weights and uses `max_weight_matching`. This logic seems to be faulty in this example: 1/1+1/4>1/2+1/2.

### PR DESCRIPTION
**Pull request by dschult**
Fixes #5388 

Add test and then fix code and docs.

Discussion point:  I think we need to remove the `maxcardinality` parameter because it doesn't make sense in a minimum weight matching to allow edges to not be included. You would return a matching with no edges. That would certainly minimize the edge weights.  Inside `min_weight_matching` we should always call `max_weight_matching` with `maxcardinality=True`.

I think this keyword should just be removed. But I deprecated it -- only instead of a warning it raises a NetworkXError telling the user that the concept doesn't make sense and that the parameter would be removed in v3.  But I still think it should just be removed now -- with a note in the release notes.

Thoughts?

### CODE REVIEW

**Comment by mjschwenne** [comment]
Reading about this bug reminded me of #4836 since they are both a similar issue with converting a maximal algorithm to take the minimum. While I haven't had the time yet to work on that issue, it is still on my mind. I wonder if other functions which take a minimal approach using a maximal algorithm have this issue. It might be a good idea to check if these are part of a large problem. I *should* start to get some free time in the coming week and try to compile a list to test.

**Review by rossbar — COMMENTED** [review]
LGTM thanks @dschult - just a couple docstring nits!

Also I think this should include a release note and a bullet in `doc/developer/deprecations` so we don't forget to remove the parameter for 3.0!

**Comment by dschult** [comment]
Sorry this took so long...  I've been worried whether adding 1 to each edge weight to avoid an edge with zero weight would change the max_weight_matching by favoring matchings with more edges in them. It took me a little bit to figure out that all the matchings that we compare in `max_weight_matching` using option `maxcardinality=True` have the same number of edges (cardinality). So adding a constant to the edge weights doesn't affect which matching is optimal. I switched from adding 1e-6 to adding 1 because it would allow integer edge weights to still be integer values. 

I've fixed the documentation errors/suggestions, so I think this is ready for re-review.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ repos:
     hooks:
     - id: black
 -   repo: https://github.com/asottile/pyupgrade
-    rev: v2.29.1
+    rev: v2.31.1
```
Comment (rossbar) [inline .pre-commit-config.yaml:10]: I'm not sure if this should get squashed into this PR or might be better to have in its own or as part of a general maintenance PR. Not a big deal, just thinking out loud!

**Review by rossbar — APPROVED** [review]
LGTM, thanks @dschult !

**Review by mjschwenne — APPROVED** [review]
Docstring explanations about the new method was informative and everything looks good!

**Comment by rossbar** [comment]
Thanks @dschult & @mjschwenne !

**Comment by mjschwenne** [comment]
Any reason your approval is green and mine is gray?

**Comment by dschult** [comment]
[Looks like](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews)  green checks are from people with write access to the repository while gray checks are from people without write access.  Is the difference in write access intentional or how is it determined?

**Comment by MridulS** [comment]
I think it's because @mjschwenne is currently part of `triage` team, and that team can move around issues, close them but they can't merge PRs into the `main` branch.

**Pull request merged**